dace/transformation/passes/dead_dataflow_elimination.py

```diff
@@ -57,15 +58,19 @@ def apply_pass(self, sdfg: SDFG, pipeline_results: Dict[str, Any]) -> Optional[D
         :return: A dictionary mapping states to removed data descriptor names, or None if nothing changed.
         """
         # Depends on the following analysis passes:
-        #  * State reachability
-        #  * Read/write access sets per state
-        reachable: Dict[SDFGState, Set[SDFGState]] = pipeline_results['StateReachability'][sdfg.cfg_id]
-        access_sets: Dict[SDFGState, Tuple[Set[str], Set[str]]] = pipeline_results['AccessSets'][sdfg.cfg_id]
+        #  * Control flow block reachability
+        #  * Read/write access sets per block
+        sdfg = region if isinstance(region, SDFG) else region.sdfg
+        reachable: Dict[ControlFlowBlock, Set[ControlFlowBlock]] = pipeline_results[
+            ap.ControlFlowBlockReachability.__name__
+        ][region.cfg_id]
+        access_sets: Dict[ControlFlowBlock, Tuple[Set[str], Set[str]]] = pipeline_results[ap.AccessSets.__name__]
         result: Dict[SDFGState, Set[str]] = defaultdict(set)
 
-        # Traverse SDFG backwards
+        # Traverse region backwards
         try:
-            state_order = list(cfg.blockorder_topological_sort(sdfg))
+            state_order: List[SDFGState] = list(cfg.blockorder_topological_sort(region, recursive=False,
+                                                                                ignore_nonstate_blocks=True))
```

Comment:
How does ignore_nonstate_blocks interact with this pass? Does it not break it for some corner cases where you have conditionals and dead dataflow?